pkg/util/xds/callbacks_chain.go

```diff
@@ -15,11 +13,13 @@ var _ envoy_xds.Callbacks = CallbacksChain{}
 
 // OnStreamOpen is called once an xDS stream is open with a stream ID and the type URL (or "" for ADS).
 // Returning an error will end processing and close the stream. OnStreamClosed will still be called.
-func (chain CallbacksChain) OnStreamOpen(ctx context.Context, streamID int64, typ string) (errs error) {
+func (chain CallbacksChain) OnStreamOpen(ctx context.Context, streamID int64, typ string) error {
 	for _, cb := range chain {
-		errs = multierr.Append(errs, cb.OnStreamOpen(ctx, streamID, typ))
+		if err := cb.OnStreamOpen(ctx, streamID, typ); err != nil {
+			return err
```

Comment:
callbacks execution now breaks after the first error. Otherwise we cannot have auth callback.